backend/YouTubeV2.Application/Services/SubscriptionService.cs

```diff
@@ -19,63 +13,52 @@ public class SubscriptionService : ISubscriptionService
         private readonly IBlobImageService _blobImageService;
         private readonly YTContext _context;
         private readonly UserManager<User> _userManager;
+
         public SubscriptionService(IBlobImageService blobImageService, YTContext context, UserManager<User> userManager)
         {
             _blobImageService = blobImageService;
             _context = context;
             _userManager = userManager;
         }
 
-        public async Task<UserSubscribtionListDto> GetSubscriptionsAsync(Guid Id, CancellationToken cancellationToken)
-        {
-            return new UserSubscribtionListDto( await _context.Subscriptions.
-                Where(s => s.SubscriberId == Id.ToString()).
-                Select(s => new SubscribtionDto(new Guid(s.SubscribeeId), _blobImageService.GetProfilePicture(s.Subscribee.Id), s.Subscribee.UserName)).
-                ToListAsync(cancellationToken));
-        }
+        public async Task<UserSubscriptionListDto> GetSubscriptionsAsync(string id, CancellationToken cancellationToken = default) =>
+            new UserSubscriptionListDto (await _context
+                .Subscriptions
+                .Include(subscribtion => subscribtion.Subscribee)
+                .Where(s => s.SubscriberId == id)
+                .Select(s => new SubscriptionDto(s.SubscribeeId, _blobImageService.GetProfilePicture(s.Subscribee.Id), s.Subscribee.UserName!))
+                .ToArrayAsync(cancellationToken));
 
-        public async Task<int> GetSubscriptionCount(Guid id, CancellationToken cancellationToken)
-        {
-            var userId = id.ToString();
+        public async Task<int> GetSubscriptionCount(string id, CancellationToken cancellationToken = default) =>
+            await _context.Subscriptions.CountAsync(s => s.SubscribeeId == id, cancellationToken);
 
-            return await _context.Subscriptions.CountAsync(s => s.SubscribeeId == userId, cancellationToken);
-        }
-        public async Task PostSubscriptionsAsync(Guid subscribeeGuid, Guid subscriberGuid, CancellationToken cancellationToken)
+        public async Task AddSubscriptionAsync(string subscribeeId, string subscriberId, CancellationToken cancellationToken = default)
         {
+            var subscribee = await _userManager.FindByIdAsync(subscribeeId)
+                ?? throw new NotFoundException($"User with id {subscribeeId} you want to subscribe not found");
```

Comment:
according to spec we shouldnt throw it but i guess it (spec) will change

Reply:
when it comes to error handling and status codes returned imo we shouldnt strictly follow the specification because it doesnt really matter. If something failed than it failed and giving different status code gives only the inforramtion what failed (i dont believe that a cross test check where we want to subscibe a non existent user and expect status code 400 as response will ever be created)